ee/clickhouse/queries/breakdown_props.py

```diff
@@ -94,3 +98,53 @@ def get_breakdown_event_prop_values(
     )
 
     return top_elements_array
+
+
+def _format_all_query(team_id: int, filter: Filter, **kwargs) -> Tuple[str, Dict]:
+    entity = kwargs.pop("entity", None)
+    parsed_date_from, parsed_date_to, date_params = parse_timestamps(
+        filter=filter, team_id=team_id, table="all_events."
+    )
+
+    props_to_filter = [*filter.properties]
+
+    if entity and isinstance(entity, Entity):
+        props_to_filter = [*props_to_filter, *entity.properties]
+
+    prop_filters, prop_filter_params = parse_prop_clauses(
+        props_to_filter, team_id, prepend="all_cohort_", table_name="all_events"
+    )
+    query = """
+            SELECT DISTINCT distinct_id, 0 as value
+            FROM events all_events
+            WHERE team_id = {} {} {} {}
+            """.format(
+        team_id, parsed_date_from, parsed_date_to, prop_filters
+    )
+    return query, {**date_params, **prop_filter_params}
+
+
+def format_breakdown_cohort_join_query(team_id: int, filter: Filter, **kwargs) -> Tuple[str, List, Dict]:
+    entity = kwargs.pop("entity", None)
+    cohorts = Cohort.objects.filter(team_id=team_id, pk__in=[b for b in filter.breakdown if b != "all"])
+    cohort_queries, params = _parse_breakdown_cohorts(cohorts)
+    ids = [cohort.pk for cohort in cohorts]
+    if "all" in filter.breakdown:
+        all_query, all_params = _format_all_query(team_id, filter, entity=entity)
+        cohort_queries.append(all_query)
+        params = {**params, **all_params}
+        ids.append(0)
+    return " UNION ALL ".join(cohort_queries), ids, params
+
+
+def _parse_breakdown_cohorts(cohorts: BaseManager) -> Tuple[List[str], Dict]:
+    queries = []
+    params: Dict[str, Any] = {}
+    for idx, cohort in enumerate(cohorts):
+        person_id_query, cohort_filter_params = format_filter_query(cohort, idx)
+        params = {**params, **cohort_filter_params}
+        cohort_query = person_id_query.replace(
+            "SELECT distinct_id", "SELECT distinct_id, {} as value".format(cohort.pk)
```

Comment:
newer Python versions FTW.

```suggestion
            f"SELECT distinct_id", "SELECT distinct_id, {cohort.pk} as value"
```